.github/workflows/lint-test.yaml

```diff
@@ -191,3 +191,31 @@ jobs:
             !/tmp/.tmp*/**/db/
           retention-days: 1
 
+  forge-unittests:
```

Comment:
I'm a bit confused about what's going on in this job, could you add some comments? Mostly about the part where we get all the commits + diffs and then how that feeds into the tests.

Reply:
That is to determine if we should run the tests, the step naming i think is pretty self explanatory but i can add comment as well